third_party/amd/python/triton_amd.cc

```diff
@@ -297,4 +309,11 @@ void init_triton_amd(py::module &&m) {
       arg.addAttr(llvm::Attribute::InReg);
     }
   });
+
+  m.def("link_hsaco",
+        [](const std::string &inPath, const std::string &outPath) {
+          if (!lldInvoke(inPath.c_str(), outPath.c_str()))
+            throw std::runtime_error("LLD failed to link hsaco source" +
+                                     inPath + " into object file" + outPath);
```

Comment:
Need to leave a space before the paths? Also I wonder whether we can decode the `s.retCode` and print out the error reason.

Reply:
the errs, whatever they might be within the lld code itself, are already being dumped to `stderr` right? 

```c++
lld::lldMain(args, llvm::outs(), llvm::errs(),
                               {{lld::Gnu, &lld::elf::link}});
```

I could capture them and ship them back through the exception? Retcode I think would be harder/mumbo-jumbo?

Reply:
I see. Missed that part. LGTM then!